pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -59,35 +58,35 @@ def reset(self) -> None:
         """Resets the loop state."""
         self._outputs = []
 
-    def on_run_start(self, batch: Any, batch_idx: int) -> None:  # type: ignore[override]
+    def on_run_start(self, kwargs: OrderedDict) -> None:  # type: ignore[override]
         """Splits the data into tbptt splits.
 
         Args:
-            batch: the current batch to run the trainstep on
-            batch_idx: the index of the current batch
+            kwargs: the kwargs passed down to the hooks.
         """
-        void(batch_idx)
+        batch = kwargs["batch"]
```

Comment:
how about if the key is missing, shall we rather?
```suggestion
        batch = kwargs.get("batch")
```

Reply:
No, that would be a silent error. I want it to raise a KeyError in this case because it would clearly show a bug in the implementation